### DISCUSSION TITLE
Predefined node-locked license and auto-create customer

### DISCUSSION THREAD

**Discussion by HappyPathAB** [Labs64/NetLicensing-Community#83] [Questions] [answered]
The [Node-Locked license model](https://netlicensing.io/wiki/node-locked) wiki page states in the section for Predefined mode that _Since the secret must be stored in a license before the first validation, this mode can’t be combined with “Auto-create Licensee” mode_. Similarly, the UI page for a node-locked module says in the tooltip for the Predefined mode that it _can't be used if Licensee Auto-create mode is enabled for the product_.

Is this accurate?

`Auto-create licensee` (or customer) is a property of the product. But a product may have more than one module, and the customer will be automatically provided licenses for only those modules which are specified as `Default` for the product.

So I would have thought that it's the `Default` setting of the module, not the `Auto-create` setting of the product, that would determine whether or not the Predefined mode can be used.

**Comment by r-brown** [comment, marked as answer]
Hello @HappyPathAB please allow me to clarify on that.

`Product . Auto-create Licensee` option creates a new customer upon first validation is this customer doesn't exist yet
`Default` flag for the product module package adds given package/licenses to the Customer on create
- this can be a trigger from the first bullet point
- or manual customer create via the management console
- or [customer creation](https://netlicensing.io/wiki/licensee-services#create-licensee) using RESTful API

So, all these default licenses not necessary exclusively related to the `Auto-create Licensee` option ON.

---

The idea behind `Predefined` mode for the [Node-Locked licensing model](https://netlicensing.io/wiki/node-locked) is to give vendors more control of how end-devices (nodes) are being used.
Having this mode available for the _auto-created-licensees_ would defeat the purpose here.
And this is exactly how other option `Client` is working.

---

I hope this clarifies your question. Please feel free to rewert on this in case of any concerns or additional clarification needed.
